src/Tools/ExternalAccess/VisualDiagnostics/Internal/VisualDiagnosticsServiceFactory.cs

```diff
@@ -0,0 +1,294 @@
+/// for debugger process referencing Maui.Essentials.dll
+/// </summary>
+[ExportCSharpVisualBasicLspServiceFactory(typeof(OnInitializedService)), Shared]
+internal sealed class VisualDiagnosticsServiceFactory : ILspServiceFactory
+{
+    private readonly LspWorkspaceRegistrationService _lspWorkspaceRegistrationService;
+    private readonly Lazy<IBrokeredDebuggerServices> _brokeredDebuggerServices;
+
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    [ImportingConstructor]
+    public VisualDiagnosticsServiceFactory(
+        LspWorkspaceRegistrationService lspWorkspaceRegistrationService,
+        Lazy<IBrokeredDebuggerServices> brokeredDebuggerServices)
+    {
+        _lspWorkspaceRegistrationService = lspWorkspaceRegistrationService;
+        _brokeredDebuggerServices = brokeredDebuggerServices;
+    }
+
+    public ILspService CreateILspService(LspServices lspServices, WellKnownLspServerKinds serverKind)
+    {
+        var lspWorkspaceManager = lspServices.GetRequiredService<LspWorkspaceManager>();
+        return new OnInitializedService(lspServices, lspWorkspaceManager, _lspWorkspaceRegistrationService, _brokeredDebuggerServices);
+    }
+
+    private class OnInitializedService : ILspService, IOnInitialized, IObserver<HotReloadNotificationType>, IDisposable
+    {
+        private readonly LspServices _lspServices;
+        private readonly LspWorkspaceManager _lspWorkspaceManager;
+        private readonly LspWorkspaceRegistrationService _lspWorkspaceRegistrationService;
+        private readonly Lazy<IBrokeredDebuggerServices> _brokeredDebuggerServices;
+        private readonly ConditionalWeakTable<Workspace, IVisualDiagnosticsLanguageService> _visualDiagnosticsLanguageServiceTable;
+        private readonly System.Timers.Timer _timer;
+        private readonly SemaphoreSlim _mutex = new SemaphoreSlim(1);
+        private List<ProcessInfo> _debugProcesses;
+        private CancellationToken _cancellationToken;
+        private IDisposable? _adviseHotReloadSessionNotificationService;
+
+        public OnInitializedService(LspServices lspServices, LspWorkspaceManager lspWorkspaceManager, LspWorkspaceRegistrationService lspWorkspaceRegistrationService, Lazy<IBrokeredDebuggerServices> brokeredDebuggerServices)
+        {
+            _lspServices = lspServices;
+            _lspWorkspaceManager = lspWorkspaceManager;
+            _lspWorkspaceRegistrationService = lspWorkspaceRegistrationService;
+            _brokeredDebuggerServices = brokeredDebuggerServices;
+            _timer = new System.Timers.Timer();
+            _timer.Interval = 1000;
+            _timer.Elapsed += Timer_Elapsed;
+            _visualDiagnosticsLanguageServiceTable = new();
+            _debugProcesses = new List<ProcessInfo>();
+        }
+
+        private void Timer_Elapsed(object sender, System.Timers.ElapsedEventArgs e)
+        {
+            _timer.Stop();
+            _ = OnTimerElapsedAsync();
```

Comment:
This needs to use `IAsynchronousOperationListener` since the life of the asynchronous operation is not captured by the caller.